### PR TITLE
Travis Deployment

### PR DESCRIPTION
**Pull request by ggetz**
Deploys results of Travis build process to S3. It is available at the url http://cesium-dev.s3-website-us-east-1.amazonaws.com/cesium/BRANCH_NAME/.

So this branch is available at http://cesium-dev.s3-website-us-east-1.amazonaws.com/cesium/travis-deploy/

You must configure your Travis settings for the repo by adding two environment variables: `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`. Your credentials must have read, write, and list permissions for the cesium-dev bucket. 

If you do not have proper credentials, the build will not be deployed, but the Travis build will still pass.

### CODE REVIEW

**Comment by mramato** [comment]
Glorious.

@ggetz can you right up an issue for us to add a travis section to the build guide with any relevant CI information.  This way we don't lost any knowledge and if someone has their own fork they can leverage what we've already done if desired.

**Comment by mramato** [comment]
It looks like the last deploy failed because of credentials errors.

**Comment by ggetz** [comment]
The last deploy came from the PR, and was triggered by `AnalyticalGraphicsInc/cesium` rather than `ggetz/cesium`. We can add credentials to the AnalyticalGraphicsInc Travis setting and it should work as well.

**Comment by mramato** [comment]
That's what I assumed.  Can you please take care of that and then we'll merge.

**Comment by ggetz** [comment]
I added the credentials to the account and restarted the build twice, but I'm still getting an error about missing credentials. I tested the deploy task locally with those credentials and that worked fine.

**Comment by mramato** [comment]
Okay, I'll take a look.

**Comment by ggetz** [comment]
Thanks.

**Comment by mramato** [comment]
I pushed this same exact branch to a branch of this repository and everything worked: http://cesium-dev.s3-website-us-east-1.amazonaws.com/cesium/test-deplo/

I'm thinking there is something going on here in how travis manages environment variables.

**Comment by mramato** [comment]
I found the answer: https://docs.travis-ci.com/user/pull-requests#Security-Restrictions-when-testing-Pull-Requests

This makes total sense, forks aren't allowed access to travis environment variables because they would then be able to read them and do whatever they want with the key (which is bad).  Security is hard but this is the correct behavior on travis' end.

@ggetz can you add a check to see if AWS had credentials and spit out a friendly message.  This link has instructions: http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html specifically, the `getCredentials` function.

**Comment by ggetz** [comment]
> can you add a check to see if AWS had credentials and spit out a friendly message.

Should the task still fail?

**Comment by mramato** [comment]
> Should the task still fail?

I would think no.

**Comment by ggetz** [comment]
Updated.

File: gulpfile.js [hunk 1]
```diff
@@ -335,12 +335,23 @@ gulp.task('deploy-s3', function(done) {
 
 });
 
+function getCredentials() {
```
Comment (mramato) [inline gulpfile.js]: This function can be replaced with 
```javascript
var getCredentials = Promise.promisify(aws.config.getCredentials, {context: aws.config})
```

You might not need the context options, it depends on how config is implemented.

**Comment by mramato** [comment]
One more comment then I'll merge

File: gulpfile.js (hunk 1 above)
Comment (ggetz) [inline gulpfile.js]: I had tried `promisify` without the context object and got errors even when I had the credentials, so I think the context is necessary.

**Comment by ggetz** [comment]
Fixed.

**Comment by mramato** [comment]
Awesome.

**Pull request merged**